api/v1alpha1/gitcommitstatus_types.go

```diff
@@ -0,0 +1,228 @@
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	Key string `json:"key"`
+
+	// Description is a human-readable description of this validation that will be shown in the SCM provider
+	// (GitHub, GitLab, etc.) as the commit status description.
+	// If not specified, defaults to empty string.
+	// +optional
+	Description string `json:"description,omitempty"`
+
+	// ValidateCommit specifies which commit SHA to validate with the expression.
+	// - "active": Validates the currently active/deployed commit (default behavior)
+	// - "proposed": Validates the proposed commit that will be promoted
+	//
+	// The validation result is always reported on the PROPOSED commit (for gating), but this field
+	// controls which commit's data is used in the expression evaluation.
+	//
+	// Examples:
+	//   validateCommit: "active" - "Don't promote if a revert commit is detected"
+	//   validateCommit: "proposed" - "Don't promote unless new commit follows naming convention"
+	//
+	// +optional
+	// +kubebuilder:default="active"
+	// +kubebuilder:validation:Enum=active;proposed
+	ValidateCommit string `json:"validateCommit,omitempty"`
+
+	// Expression is evaluated using the expr library (github.com/expr-lang/expr) against commit data
+	// for environments in the referenced PromotionStrategy.
+	// The expression must return a boolean value where true indicates the validation passed.
+	//
+	// The commit validated is determined by the ValidateCommit field:
+	// - "active" (default): Validates the ACTIVE (currently deployed) commit
+	// - "proposed": Validates the PROPOSED commit (what will be promoted)
+	//
+	// The validation result is always reported on the PROPOSED commit to enable promotion gating.
+	//
+	// Use Cases by Mode:
+	//   Active mode: State-based gating - validate current environment before allowing promotion
+	//     - "Don't promote until active commit has required sign-offs"
+	//     - "Verify current deployment meets compliance before next promotion"
+	//     - "Ensure active commit is not a revert before promoting"
+	//
+	//   Proposed mode: Change-based gating - validate the incoming change itself
+	//     - "Don't promote unless new commit follows naming convention"
+	//     - "Ensure proposed commit has proper JIRA ticket reference"
+	//     - "Require specific author for proposed changes"
+	//
+	//
+	// Available variables in the expression context:
+	//   - Commit.SHA (string): the commit SHA being validated (active or proposed based on ValidateCommit)
+	//   - Commit.Subject (string): the first line of the commit message
+	//   - Commit.Body (string): the commit message body (everything after the subject line)
+	//   - Commit.Author (string): commit author email address
+	//   - Commit.Committer (string): committer email address
+	//   - Commit.Trailers (map[string][]string): git trailers parsed from commit message
+	//
+	// +required
+	Expression string `json:"expression"`
```

Comment:
Should we have it return a structure so that we can expand features in the future?

```yaml
expression: |
  {
    "passed": 1 == 2,
    "message": Commit.SHA + " is busted"
  }
```

Or maybe that would just end up being a new field:

```yaml
messageExpression: |-
  Commit.SHA + " is busted"
```